plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchHelper.groovy

```diff
@@ -199,22 +200,25 @@ class AwsBatchHelper {
      *      is made of multiple *page* this method returns only the first one
      */
     String getTaskLogStream(String jobId) {
-        final streamId = getLogStreamId(jobId)
-        if( !streamId ) {
+        final containerDetail = getContainerDetails(jobId)
+        if( !containerDetail.logStreamName ) {
             log.debug "Unable to find CloudWatch log stream for batch job id=$jobId"
             return null
         }
+        // Fallback to the default log group name if log group name is not set
+        final logGroupName = containerDetail.logConfiguration?.options?.get('awslogs-group')
+            ?: '/aws/batch/job'
 
         final logRequest = new GetLogEventsRequest()
-                .withLogGroupName("/aws/batch/job")
-                .withLogStreamName(streamId)
+                .withLogGroupName(logGroupName)
```

Comment:
If I'm understanding correctly, the log group should match the one define in the config 

https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-amazon/src/main/nextflow/cloud/aws/config/AwsBatchConfig.groovy#L62-L65

Reply:
It should be yes, but I thought it would be easier to pull directly from the container response since we're already grabbing that information.

I can change it to pass in the log group from the config file in the `getTaskLogStream` method?